docs/testplan/dual_tor/dual_tor_active_active_test_plan_action_items.md

```diff
@@ -0,0 +1,382 @@
+# active-active dualtor test plan action items
+This document is served as a blueprint to briefly descibe the roadmap to support active-active dualtor tests and list out all possible active-active test items.
+This document is subject to change as the project evolves.
+**NOTE**: the scope/detail of server NiC simulator is not within the scope of this document.
+
+## test infrastructure roadmap
+1. Add topo definition changes to support per-port cable types:
+   - `active-standby`
+   - `active-active`
+2. Implement server NiC simulator(please refer the server NiC simulator doc for more detail)
+3. Add testbed deployment support: `add-topo`, `remove-topo`, `restart-ptf`, and `deploy-mg`.
+    3.1. minigraph generation/parsing support for the introduction of `active-active` cable type.
+    3.2. support starting either server NiC simulator or mux simulator selectively based on the port cable type.
+    3.3. changes to `vm_topology` to configure ovs bridge based on the port cable type.
+4. Add utilities modules/functions to interact with the server NiC simulator.
+5. Add sanity check support to verify the `active-active` port and its corresponding server NiC simulator service health.
+
+## test infrastrucure items
+The test infrastructure items listed here serves the following purposes:
+- common utilities to interacts with server NiC simulator service that runs on the test server.
+- common utilities to retrieve ToRs' or ports' related data including port mapping, port forwarding state, DB state, etc.
+- common utilities to interact with ToRs to change specific settings(forwarding state, port state, etc).
+- common utilities to send/verify traffic(Vxlan tunnel traffic checker).
+
+**NOTE**: the original dualtor project had already implemented some of the test infrastructure items, those parts could simply be reused.
+
+### server nic related items
+```python
+@pytest.fixture(scope="session")
+def nic_simulator_server_info(request, tbinfo):
+    """Return a tuple of (nic simulator IP, nic simulator port, vmset name)."""
+    pass
+
+@pytest.fixture(scope="session")
+def restart_nic_simulator(nic_simulator_server_info, vminfo):
+    """Restart server nic simulator."""
+    pass
+
+@pytest.fixture(scope='module')
+def url(mux_server_url, duthost, tbinfo):
```

Comment:
1. The nic simulator may be listening on a different port. Not sure if we can reuse the mux_server_url here.
2. Is name `url` taken? Probably need a longer name to avoid potential conflicts.